python_modules/libraries/lakehouse/lakehouse/house.py

```diff
@@ -184,11 +201,30 @@ def pipeline_multi_deps():
         else:
             input_mappings = None
 
+        end_solid = SolidDefinition(
+            name="__lakehouse_end",
+            input_defs=[
+                InputDefinition("{}__end".format("__".join(solid_name)), Nothing)
+                for solid_name in solid_defs
+            ],
+            output_defs=[OutputDefinition(Nothing)],
+            compute_fn=lambda *args: None,
+        )
+        solid_deps[end_solid.name] = {
+            "{}__end".format(solid_def.name): DependencyDefinition(solid_def.name, "end")
+            for solid_def in solid_defs.values()
+        }
+        solid_defs[end_solid.name] = end_solid
+
+        nothing_output = OutputDefinition(Nothing, "end")
+        output_mappings = [nothing_output.mapping_from(end_solid.name)]
```

Comment:
It's a bummer that you have to add a no-compute solid here to pull this off - but I don't think there is a way around that with the current abstractions.

One thing you can do to make this a bit cleaner I think is make to use `MultiDependencyDefinition` to fan in all the outputs to a single `Nothing` input. If you did you may be able to drop [1] and just fan in all the real output_defs in to it.

https://docs.dagster.io/_apidocs/pipeline#dagster.MultiDependencyDefinition

Reply:
I'm also a little wary of adding this no-compute solid for all cases.  It seems like it could slow down pipelines, and what if someone wants a solid to run after one of the lakehouse solids, but not wait for the other ones?

I'm not an expert on composite solids, but would it make sense to expose a composite solid output for each of the lakehouse solids?  Then the calling code could decide which ones to depend on?